### PR TITLE
gtest: INSTANTIATE_TEST_CASE_P is deprecated

### PR DESCRIPTION
**Pull request by tswhison**
Replace with INSTANTIATE_TEST_SUITE_P, per the gtest docs.
Set the gtest variables correctly when manually fetching gtest.

Link with debug gtest libraries when CMAKE_BUILD_TYPE is Debug or
RelWithDebInfo.

Fix bug in object_api tests.  Reset cleanup_size to 0 at the beginning
of each test, because main() is called multiple times.

Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

### CODE REVIEW

**Review by anandaravuri — APPROVED** [review]

File: cmake/modules/OPAE.cmake [hunk 1]
```diff
@@ -57,7 +57,7 @@ include(FindUdev)
 
 include(OPAECompiler)
 if(OPAE_BUILD_TESTS)
-    find_package(GTest 1.8.0)
+    find_package(GTest 1.11.0)
```
Comment (r-rojo) [inline cmake/modules/OPAE.cmake:60]: Should we store this in a cmake variable? That way when we change versions we only have to change it in one place?

File: tests/framework/mock/opae_fixtures.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+#include "opae_fixtures.h"
+
+namespace opae {
+namespace testing {
+
+uint32_t opae_p::num_tokens_for(fpga_properties filter)
```
Comment (r-rojo) [inline tests/framework/mock/opae_fixtures.cpp:31]: What is the intent of this class?

File: cmake/modules/OPAE.cmake (hunk 1 above)
Comment (tswhison) [reply cmake/modules/OPAE.cmake:60]: Where should we store it?

File: tests/framework/mock/opae_fixtures.cpp (hunk 2 above)
Comment (tswhison) [reply tests/framework/mock/opae_fixtures.cpp:31]: I'm intending for it to replace most of our test fixtures which, today, contain boilerplate that is replicated over and over.

File: cmake/modules/OPAE.cmake (hunk 1 above)
Comment (tswhison) [reply cmake/modules/OPAE.cmake]: done

**Review by r-rojo — APPROVED** [review]

**Pull request merged**